api/src/main/java/org/infinispan/api/reactive/KeyValueEntry.java

```diff
@@ -0,0 +1,50 @@
+package org.infinispan.api.reactive;
+
+import java.util.Objects;
+
+public class KeyValueEntry<K, V> {
```

Comment:
where is this be used?
should it implement `Map.Entry`?

Reply:
This is used in ContinuousQuert for example as the return of the publisher.
but this part is not clear enough yet (typed vs object). I think it needs a second thought after this PR is merged